tests/test_s3api_policy_bucket.sh

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env bats
+
+# Copyright 2024 Versity Software
+# This file is licensed under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+test_s3api_policy_delete_bucket_policy() {
+  policy_file="policy_file"
+  username=$USERNAME_ONE
+  password=$PASSWORD_ONE
+
+  run create_test_file "$policy_file" 0
+  assert_success
+
+  effect="Allow"
+  principal="$username"
+  action="s3:DeleteBucketPolicy"
+  resource="arn:aws:s3:::$BUCKET_ONE_NAME"
+
+  setup_user "$username" "$password" "user" || fail "error creating user"
+
+  run setup_bucket "s3api" "$BUCKET_ONE_NAME"
+  assert_success
+
+  if delete_bucket_policy_with_user "$BUCKET_ONE_NAME" "$username" "$password"; then
+    fail "able to delete bucket policy with user $username without right permissions"
+  fi
+  setup_policy_with_single_statement "$TEST_FILE_FOLDER/$policy_file" "dummy" "$effect" "$principal" "$action" "$resource" || fail "failed to set up policy"
```

Comment:
Any reason this, and the commands below it, aren't wrapped in a `run`?

Reply:
These are old, unconverted functions.  Just broke up a large file, and didn't change some of the fails to run/assserts in it yet.  I've gradually been doing this commit by commit.  Can change some more of these ones now if you want though.